cpp/daal/src/algorithms/kernel_function/polynomial/kernel_function_polynomial_dense_default_impl.i

```diff
@@ -0,0 +1,243 @@
+{
+    SafeStatus safeStat;
+
+    char trans   = 'T';
+    char notrans = 'N';
+
+    const size_t nFeatures = a1->getNumberOfColumns();
+    const size_t nVectors1 = a1->getNumberOfRows();
+    const size_t nVectors2 = a2->getNumberOfRows();
+
+    algorithmFPType alpha  = (algorithmFPType)(par->scale);
+    algorithmFPType beta   = 0.0;
+    algorithmFPType shift  = (algorithmFPType)(par->shift);
+    algorithmFPType degree = (algorithmFPType)(par->degree);
+    const size_t d         = par->degree;
+
+    const bool isSOARes = r->getDataLayout() & NumericTableIface::soa;
+
+    DAAL_OVERFLOW_CHECK_BY_ADDING(size_t, nVectors1, nVectors2);
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nVectors1 + nVectors2, sizeof(algorithmFPType));
+
+    const size_t blockSize = 128;
+    const size_t nBlocks1  = nVectors1 / blockSize + !!(nVectors1 % blockSize);
+    const size_t nBlocks2  = nVectors2 / blockSize + !!(nVectors2 % blockSize);
+
+    TlsMem<algorithmFPType, cpu> tlsMklBuff(blockSize * blockSize);
+
+    daal::conditional_threader_for((nBlocks1 > 2), nBlocks1, [&, isSOARes](const size_t iBlock1) {
+        DAAL_INT nRowsInBlock1 = (iBlock1 != nBlocks1 - 1) ? blockSize : nVectors1 - iBlock1 * blockSize;
+        DAAL_INT startRow1     = iBlock1 * blockSize;
+
+        ReadRows<algorithmFPType, cpu> mtA1(*const_cast<NumericTable *>(a1), startRow1, nRowsInBlock1);
+        DAAL_CHECK_BLOCK_STATUS_THR(mtA1);
+        const algorithmFPType * const dataA1 = const_cast<algorithmFPType *>(mtA1.get());
+
+        WriteOnlyRows<algorithmFPType, cpu> mtRRows;
+        if (!isSOARes)
+        {
+            mtRRows.set(r, startRow1, nRowsInBlock1);
+            DAAL_CHECK_MALLOC_THR(mtRRows.get());
+        }
+
+        daal::conditional_threader_for((nBlocks2 > 2), nBlocks2, [&, nVectors2, nBlocks2](const size_t iBlock2) {
+            DAAL_INT nRowsInBlock2 = (iBlock2 != nBlocks2 - 1) ? blockSize : nVectors2 - iBlock2 * blockSize;
+            DAAL_INT startRow2     = iBlock2 * blockSize;
+
+            ReadRows<algorithmFPType, cpu> mtA2(*const_cast<NumericTable *>(a2), startRow2, nRowsInBlock2);
+            DAAL_CHECK_BLOCK_STATUS_THR(mtA2);
+            const algorithmFPType * const dataA2 = const_cast<algorithmFPType *>(mtA2.get());
+
+            if (!isSOARes)
+            {
+                algorithmFPType * const dataR = mtRRows.get() + startRow2;
+                Blas<algorithmFPType, cpu>::xxgemm(&trans, &notrans, &nRowsInBlock2, &nRowsInBlock1, (DAAL_INT *)&nFeatures, &alpha, dataA2,
+                                                   (DAAL_INT *)&nFeatures, dataA1, (DAAL_INT *)&nFeatures, &beta, dataR, (DAAL_INT *)&nVectors2);
+
+                PRAGMA_IVDEP
+                PRAGMA_VECTOR_ALWAYS
+                for (size_t i = 0; i < nRowsInBlock1; ++i)
+                {
+                    for (size_t j = 0; j < nRowsInBlock2; ++j)
+                    {
+                        dataR[i * nVectors2 + j] += shift;
```

Comment:
shift can use how parameter `beta` in GEMM, right?

Reply:
Yes, but we have to fill result table with ones to be able to do it. And I think we won't get any performance speed up because of it